lib/flutter_secure_storage.dart

```diff
@@ -140,21 +139,21 @@ enum IOSAccessibility {
 
 class IOSOptions extends Options {
   IOSOptions(
-      {String groupId,
-      IOSAccessibility accessibility = IOSAccessibility.unlocked})
+      {String? groupId,
+      IOSAccessibility? accessibility = IOSAccessibility.unlocked})
       : _groupId = groupId,
         _accessibility = accessibility;
 
-  final String _groupId;
-  final IOSAccessibility _accessibility;
+  final String? _groupId;
+  final IOSAccessibility? _accessibility;
   @override
   Map<String, String> _toMap() {
     final m = <String, String>{};
     if (_groupId != null) {
-      m['groupId'] = _groupId;
+      m['groupId'] = _groupId!;
```

Comment:
Need to dig into null-safe dart, but why do we need  `_groupId!` and not `_groupID`?

Reply:
I've been using null-safe Dart for a while, and situations like this seem to come up a lot. The inference isn't great.

Reply:
With sound null safety, I don't think there are any performance hits though.

Reply:
In this case, it's because the code is using a member of the instance, and it could conceivably change or be implemented by a getter that changes the returned value all the time.

So, the analyzer can't be sure that `_groupId` tested in the `if` statement will have the same value by the time we get to the next line.

Solutions — there are discussions in the Dart issue tracker on making the analyzer smarter, or making there be a difference between getters and instance members. But, in the near term, the solution is either to use the `!` or to assign a local variable, instead of using the instance member.

Reply:
That's true, even though not a single person in the known universe would ever implement the getter in a way that would set the value to null, but the analyzer still cannot be sure. Even though it can be 99.9999999999999% sure 😄